sdk/appconfiguration/AzureAppConfiguration/Source/AppConfigurationClient.swift

```diff
@@ -55,7 +55,7 @@ public class AppConfigurationClient: PipelineClient {
                     HeadersPolicy(),
                     UserAgentPolicy(),
                     authPolicy,
-                    ContentDecodePolicy(),
+                    ContentDecodePolicy(logger: options?.logger),
```

Comment:
I wish we could avoid passing a logger here. All the other policies get it from the request context. Maybe if we delay initializing the xmlParser and only do it lazily in onRequest, that would solve the problem. I don't see a particular advantage to initializing it inside ContentDecodePolicy's init method.

Reply:
I agree and will take a second look.

Reply:
Okay, should be fixed now.